core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -438,10 +457,15 @@ public EmailMessage moveFile(EmailMessage f, String destinationFolder, boolean c
 
 	@Override
 	public EmailMessage copyFile(EmailMessage f, String destinationFolder, boolean createFolder) throws FileSystemException {
-		ExchangeService exchangeService = getConnection();
+		ExchangeFileSystemResolver resolver = getResolver(destinationFolder);
+		ExchangeService exchangeService = getConnection(resolver.getMailbox());
+
 		try {
-			FolderId destinationFolderId = getFolderIdByFolderName(exchangeService, destinationFolder, createFolder);
-			return (EmailMessage)f.copy(destinationFolderId);
+			FolderId destinationFolderId = getFolderIdByFolderName(exchangeService, resolver, createFolder, false);
+
+			Item destinationItem = f.copy(destinationFolderId);
+
+			return EmailMessage.bind(exchangeService, destinationItem.getId());
```

Comment:
Volgens mij is bind() een dure operatie. Is het mogelijk om die alleen te doen als het nodig is?